add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt

```diff
@@ -0,0 +1,41 @@
+package dev.flutter.multipleflutters
+
+import java.lang.ref.WeakReference
+
+/**
+ * Interface for getting notifications when the DataModel is updated.
+ */
+interface DataModelObserver {
+    fun onCountUpdate(newCount : Int)
+}
+
+/**
+ * A singleton/observable data model for the data shared between Flutter and the host platform.
+ *
+ * This is the definitive source of truth for all data.
+ */
+class DataModel {
+    companion object {
+        val instance = DataModel()
+    }
+
+    private val observers =  mutableListOf<WeakReference<DataModelObserver>>()
+
+    public var counter = 0
+        set(value) {
+            field = value
+            for (observer in observers) {
+                observer.get()?.onCountUpdate(value)
```

Comment:
you want to do the same thing as iOS here? (e.g. cull the dead references)?

Reply:
though for fwiw, if you use the engine cache, then things will never be gc'ed under the weak reference. I'd probably just stick to explicit destructions which make things more manageable in the JVM anyway.

Reply:
I'm doing the same thing I'm doing on iOS.  Skip on notify, cull on removeObserver.

This should handle anybody that is being GC'd or manually deleted.